### ISSUE TITLE
[ BUG ] beforeSerialize and afterSerialize are not executed

### ISSUE THREAD

**Issue by daftfox** [GillianPerard/typescript-json-serializer#206]
Version
6.0.0

Description
When I declare a `JsonObject`, with a `beforeSerialize` and/or `afterSerialize` property and proceed to try to serialise the object in question, the custom serialisation methods are not executed.

```js
const customSerialiser = (date: Date): string => {
  console.log(date);
  return '1970-01-01';
}

@JsonObject()
class Foo {
  @JsonProperty({
    beforeSerialize: console.log,
    afterSerialize: customSerialiser
  })
  timestamp: Date;
}

const serializer = new JsonSerializer();
const foo = new Foo();
foo.timestamp = new Date();

serializer.serialize(foo);
```

Expected result:
See a log from either the `customSerialiser` method or from the direct reference to `console.log` that's passed in.

Actual result:
No logs were printed. No custom serialisation occured. Date object remains unchanged.

On which OS the bug appears?
Windows 10

What is your project type?
Angular

**Comment by GillianPerard** [comment]
Hello,

It works perfectly for me: https://stackblitz.com/edit/angular-te6zmj?file=src%2Fmain.ts
The only one problem you have in your example is the `console.log` in `beforeSerialize`, it prints properly but return `undefined` so the `afterSerialize` receives this new value.

Are you sure you enable these two properties in `tsconfig.json` : 
```
"emitDecoratorMetadata": true,
"experimentalDecorators": true,
```

**Comment by daftfox** [comment]
Hi @GillianPerard thanks for the rapid response!

Fair point re. using console.log in the `beforeSerialize` method. IDE would probably have complained about this example.
I have indeed set the compiler configuration and can also see the `beforeDeserialize` and `afterDeserialize` methods being executed.

With all that being said I just had a moment of clarity and realised that the reason the decorated methods aren't executed is because the object I'm trying to serialise isn't a class instance of Foo, but a typecasted object built from form values. DOH!

Closing since it wasn't your awesome lib, but me who was being a doofus. :)
